test/Microsoft.DotNet.Darc.Tests/DependencyCoherencyTests.cs

```diff
@@ -231,6 +231,14 @@ public async Task CoherencyUpdateTests6()
         {
             u.From.Should().Be(depA);
             u.To.Version.Should().Be("v2");
+        }, u =>
```

Comment:
Were the tests wrong before?

Reply:
I wouldn't say they were wrong, but they were weird. They were kind of behaving like a unified build:
 - They'd have two dependencies Foo and Bar, where Bar would have `CoherentParentDependency=Foo`
 - Then they'd create a list of assets (in a real scenario all of these assets are coming from the same build), and both Foo and Bar would be in them
 - In the old version of the tests, only Foo would get updated

But the thing is I believe these tests were wrong. The whole purpose of CoherentParentDependency is to update a certain dependency only when it's parent is getting an update. This is what we're expecting in Unified Build.
So I've adjusted the tests accordingly

Reply:
Ah, cool thanks